packages/ckeditor5-dev-env/lib/release-tools/tasks/bumpversions.js

```diff
@@ -0,0 +1,326 @@
+
+	// Displays packages that won't match to specified criteria and checks whether there is at least one package
+	// that should update its version.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @returns {Map.<String, ReleaseDetails>}
+	function isAnythingForRelease( packages ) {
+		logProcess( 'Checking whether is there anything for updating...' );
+
+		displaySkippedPackages( pathsCollection.skipped );
+
+		if ( packages.size === 0 ) {
+			throw new Error( 'None of the packages contains any changes since its last release. Aborting.' );
+		}
+
+		return packages;
+	}
+
+	// Asks a user whether the process should be continued.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @returns {Promise.<Map.<String, ReleaseDetails>>}
+	function confirmUpdate( packages ) {
+		logProcess( 'Should we continue?' );
+
+		return cli.confirmUpdatingVersions( packages )
+			.then( isConfirmed => {
+				if ( !isConfirmed ) {
+					throw new Error( BREAK_RELEASE_MESSAGE );
+				}
+
+				return packages;
+			} );
+	}
+
+	// Prepare a map that contains new versions of packages. It will be used for updating dependencies' version during the release.
+	//
+	// This map contains new versions of all packages that will be released and current version of all packages which won't match to
+	// specified criteria or won't be released (no changes between current and previous release).
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @returns {Object}
+	function prepareDependenciesVersions( packages ) {
+		logProcess( 'Preparing to updating versions of dependencies...' );
+
+		const dependencies = new Map();
+
+		// For every package that will be released, save its name and the latest version.
+		for ( const [ packageName, { version } ] of packages ) {
+			dependencies.set( packageName, version );
+		}
+
+		// Packages which won't match to specified criteria won't be released but we want to update their
+		// versions in packages where skipped packages are defined as dependencies.
+		for ( const packagePath of pathsCollection.skipped ) {
+			const packageJson = getPackageJson( packagePath );
+
+			dependencies.set( packageJson.name, packageJson.version );
+		}
+
+		// Even if some packages match to specified criteria but won't be releases because of no changes,
+		// we want to update their version in packages where they are defined as dependency.
+		for ( const packagePath of pathsCollection.matched ) {
+			const packageJson = getPackageJson( packagePath );
+
+			if ( !dependencies.has( packageJson.name ) ) {
+				dependencies.set( packageJson.name, packageJson.version );
+			}
+		}
+
+		return { packages, dependencies };
+	}
+
+	// Filter out packages which won't be released.
+	//
+	// @params {Map.<String, ReleaseDetails>} packages
+	// @params {Map} dependencies
+	// @returns {Object}
+	function filterPackagesThatWillNotBeReleased( packages, dependencies ) {
+		logDryRun( 'Filtering out packages that will not be updated...' );
```

Comment:
Not normal `log()`?

Reply:
I mean, you see more information what happens with the `dry run`. I think that such information can be useful in the real scenario too.

Reply:
From the real release point of view, it does not make any sense because we want to release packages that are ready to release. 

We know which packages will be releasing (because we know where the changelogs were generated) so the log "filtering out packages that will not be updated" does not make any sense because we don't display which packages were filtered out.